modules/evm/rpc/src/lib.rs

```diff
@@ -64,6 +52,23 @@ fn error_on_execution_failure(reason: &ExitReason, data: &[u8]) -> Result<()> {
 	}
 }
 
+fn decode_revert_message(data: &[u8]) -> String {
+	let invalid: String = "invalid revert message".into();
+	// A minimum size of error function selector (4) + offset (32) + string length
+	// (32) should contain a utf-8 encoded revert reason.
+	if data.len() > 68 {
+		let message_len = U256::from(&data[36..68]).saturated_into::<usize>();
+		if data.len() < message_len {
```

Comment:
should be `if data.len() < message_len + 68 {`?
and can you add some tests to `decode_revert_message` using data you get from evm